src/meta/meta_backup_service.h

```diff
@@ -97,58 +92,23 @@ struct backup_start_time
            << std::setfill('0') << std::to_string(minute);
         return ss.str();
     }
-    // NOTICE: this function will modify hour and minute, if time is invalid, this func will set
-    // hour = 24, minute = 0
+
+    // If start_time is valid, return true.
     bool parse_from(const std::string &time)
     {
-        if (::sscanf(time.c_str(), "%d:%d", &hour, &minute) != 2) {
+        if (sscanf(time.c_str(), "%d:%d", &hour, &minute) != 2) {
             return false;
-        } else {
-            if (hour > 24) {
-                hour = 24;
-                minute = 0;
-                return false;
-            }
-
-            if (hour == 24 && minute != 0) {
-                minute = 0;
-                return false;
-            }
-
-            if (minute >= 60) {
-                hour = 24;
-                minute = 0;
-                return false;
-            }
         }
-        return true;
-    }
-
-    // return the interval between new_hour:new_min and start_time,
-    // namely new_hour:new_min - start_time;
-    // unit is ms
-    int64_t compute_time_drift_ms(int32_t new_hour, int32_t new_min)
-    {
-        int64_t res = 0;
-        // unit is hour
-        res += (new_hour - hour);
-        // unit is minute
-        res *= 60;
-        res += (new_min - minute);
-        // unit is ms
-        return (res * 60 * 1000);
+        return (hour >= 0 && hour < 24 && minute >= 0 && minute < 60);
     }
 
-    // judge whether we should start backup base current time
     bool should_start_backup(int32_t cur_hour, int32_t cur_min)
     {
-        if (hour == 24) {
-            // erase the restrict of backup_start_time, just for testing
+        if (hour == 24 && minute == 0) {
+            // only for tests
```

Comment:
this is too tricky. I don't want any test code appears in such way.

Try this: https://github.com/XiaoMi/rdsn/blob/master/include/dsn/utility/fail_point.h

```
FAIL_POINT_INJECT_F(should_start_backup, []() -> bool {
  return true;
});
```

When you in test calling `fail::cfg("should_start_backup", return())`, and this func will return true.

Reply:
Actually we didn't call `should_start_backup` directly in tests, we called `issue_new_backup_unlocked` and expected starting a new backup in `policy_context_test.test_app_dropped_during_backup`.